src/amo/reducers/versions.js

```diff
@@ -24,19 +26,25 @@ export type AddonVersionType = {
 export const createInternalVersion = (
   version: ExternalAddonVersionType,
 ): AddonVersionType => {
-  // TODO: We should find the correct file, for now we'll just take the first one.
-  // Note that multiple files per version is going away, but not quite yet.
-  const file = version.files[0];
-  return {
+  const addonVersion = {
     compatibility: version.compatibility,
-    created: file.created,
-    files: version.files,
-    filesize: file.size,
+    platformFiles: createPlatformFiles(version),
     id: version.id,
     license: { name: version.license.name, url: version.license.url },
     releaseNotes: version.release_notes,
     version: version.version,
   };
+
+  const file = findFileForPlatform({
+    platformFiles: addonVersion.platformFiles,
+    userAgentInfo: state.api.userAgentInfo, // this won't work currently
```

Comment:
> 1. pass `userAgentInfo` in the action payload
> 2. call a selector function to get `created` and `filesize` when rendering the versions.

Yes, I think these are the two best options. I am torn but I guess either would be fine. 

I think there would be a negligible performance penalty if you call a selector function when rendering. Premature optimization is usually not a great idea. 

As for passing `userAgentInfo` in the action payload, this has the downside in that the app will no longer react to state changes affecting `userAgentInfo`. In practice that's probably fine because `userAgentInfo` never changes and I can't think of any reason why it would in the future.

I don't know if this helps. I think you could go in either direction 🤷‍♂️